### PR TITLE
Implemented custom emoji UI.

### LINKED ISSUE #17596: PoC: Custom UI for picking emojis
## Provide a description of the task

In #17377, we used a ready-to-use component to search for emojis and insert them into the editor.

Unfortunately, we still need to improve a few things (#17565 and #17573). On a daily, we discussed whether we could implement a custom UI, e.g., based on the special characters dialog, instead of using an external library for that.

Implementing a custom UI gives us:

* Full control over focus (<kbd>Tab</kbd> navigation),
* Design aligned to CKEditor 5 theme,
* Translated UI in languages that we support,
* An option to display the label "No found results" when no emoji was found when searching,
* More?

I would split the spike issue into three phases:

1. Based on the special characters grid:
	Image
	* Add an input for searching emojis above the `<select>` element.
	* `<select>` allows choosing a category (smiles, animals, etc.)
	* The footer section displays an emoji name.
2. Add an option to select skin (an icon next to the search input).
3. Replace `<select>` with tabs.

(2.) and (3.) could be done in any order.

Please, do not focus on the "proper" solution. We want to know if it's possible. The polishing phase will come later.

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal: Implemented custom emoji UI. Closes #17596.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by martnpaneq** [comment]
From what I can observe there are some visual glitches to the emoji picker. One icon is not displayed, and there are two random arrows behind some emojis.
image

**Comment by przemyslaw-zan** [comment]
☝️ Are you sure those are the issues of the picker itself, and not Mac specific missing emoji implementations? @martnpaneq

**Comment by martnpaneq** [comment]
Yeah, probably updating my OS will resolve the missing icon. But what about the arrow emojis being displayed behind others? 
image

File: packages/ckeditor5-emoji/src/emojipicker.ts [hunk 1]
```diff
@@ -8,21 +8,38 @@
...
 	clickOutsideHandler,
 	ContextualBalloon,
-	View,
-	ButtonView,
+	Dialog,
 	MenuBarMenuListItemButtonView
 } from 'ckeditor5/src/ui.js';
+import EmojiToneView, { type SkinToneId } from './ui/emojitoneview.js';
+
+export type EmojiGroup = {
```
Comment (martnpaneq) [inline packages/ckeditor5-emoji/src/emojipicker.ts:34]: Please move types to the bottom to be consistent with CKEditor 5 code style.

**Review by martnpaneq — APPROVED** [review]

**Pull request merged**